Runtime/Scripts/RGBotConfigs/RGState.cs

```diff
@@ -68,6 +59,73 @@ protected virtual RGStateEntity CreateStateEntity()
         {
             return new RGStateEntity();
         }
+        
+        
+        // Used to fill in the core state for any RGEntity that does NOT have an
+        // RGState implementation on its game object
+        public static RGStateEntity GenerateCoreStateForRGEntity(RGEntity rgEntity)
+        {
+            var state = new RGStateEntity();
+            PopulateCoreStateForEntity(state, rgEntity);
+            if (rgEntity.includeStateForAllBehaviours)
+            {
+                PopulateEverythingStateForEntity(state, rgEntity);
+            }
+            return state;
+        }
+
+        private static void PopulateCoreStateForEntity(RGStateEntity state, RGEntity entity)
+        {
+            var theTransform = entity.transform;
+            
+            state["id"] = theTransform.GetInstanceID();
+            state["type"] = entity.objectType;
+            state["isPlayer"] = entity.isPlayer;
+            state["isRuntimeObject"] = entity.isRuntimeObject;
+            state["position"] = theTransform.position;
+            state["rotation"] = theTransform.rotation;
+        }
+
+        private static void PopulateEverythingStateForEntity(RGStateEntity state, RGEntity entity)
+        {
+            var obsoleteAttributeType = typeof(ObsoleteAttribute);
+            // find all Components and get their values
+            var components = entity.gameObject.GetComponents<Component>();
+            foreach (var component in components)
+            {
+                // skip 'expensive' components, only get colliders and MonoBehaviours
+                if (component is Collider or Collider2D or MonoBehaviour and not RGState and not RGAction and not RGEntity and not RGAgent)
+                {
+                    var type = component.GetType();
+                    var dictionary = new Dictionary<string, object>();
+                    foreach (PropertyInfo prop in type.GetProperties())
```

Comment:
Would it be possible to find these properties _without_ using reflection? Or did we ever test if reflection in runtime builds is fully supported on the platforms our SDK supports?

https://docs.unity3d.com/Manual/ScriptingRestrictions.html

Reply:
For any arbitrary user written class, it is unlikely we could find a non-reflection solution that didn't cause them extra work (thus defeating the purpose of adding this feature in the first place).

I would be happy though to enhance the tooltip disclaimer to indicate that this option is only safe in situations where `AOT` is not in use and give them the link you pasted (https://docs.unity3d.com/Manual/ScriptingRestrictions.html)

Reply:
From the link, `AOT` looks like its supported when the backend scripting is `IL2CPP` and not when it's `Mono`. We can grab the scripting backend through an Editor script like:

```
using UnityEditor
public static string GetScriptingBackend() 
{
   BuildTargetGroup buildTarget = EditorUserBuildSettings.selectedBuildTargetGroup;
   ScriptingImplementation scriptingBackend = PlayerSettings.GetScriptingBackend(buildTargetGroup);
   return scriptingBackend.ToString(); // will return Mono or IL2CPP
}
```
We could implement a custom inspector for RGState and disable the checkbox if they're on a Mono scripting backend. 

I'm not seeing this as a blocking change though, so if you could at least add to the tooltip that'd look good to me